lib/ContentTypeParser.js

```diff
@@ -168,7 +168,7 @@ function rawBody (request, reply, options, parser, done) {
 
     var result = parser.fn(req, body, done)
     if (result && typeof result.then === 'function') {
-      result.then(body => done(null, body)).catch(err => done(err))
+      result.then(body => done(null, body), done)
```

Comment:
Errors thrown by the `done` function wont be caught anymore. Is this ok?

Reply:
When using promises you can catch errors in two ways:

Reply:
I could be misunderstanding something but the first example will also catch the errors raised by `fn` whereas the second one wont. Right? It might be that `done` it is not going to raise any error, and in that case it would be fine.

```js
Promise.resolve(1)
.then(() => { throw new Error('done() error') })
.catch(() => console.log('Caught!'))                    // Called

Promise.resolve(1)
.then(
  () => { throw new Error('done() error') },
  () => console.log('Caught!')                           // Never called
)
```

Reply:
@dicearr is right in the theoretical side. However in this case this is not needed. `done()` is set up to be called in a callback-based scenario. In any case, we can just add in a little try-catch.
Avoiding the `.catch()`  saves us the allocation of one promise.